### ISSUE TITLE
Separation of troops by entering the exact number: enhancement

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#1631]
Currently, when we try to split troops to any free slot we watch a dialog window, where we can enter any number.
Default splitting divides units 50/50, and we have the number of creatures equal to 50% in this window.
![image](https://user-images.githubusercontent.com/55348946/92810038-d708b380-f3c5-11ea-8fdf-3970f7405909.png)
To change the quantity of creatures we can start entering digits from keyboard. But this would add to the current (50%) number the digit we pressed from the right side. 
Ex: if we press "3", the quantity on the screencshot would turn into "153. If we start doing this in a default window (50\50) the final number exceeds the total creatures ammount - and final ammount would turn into maximum available. Which can be set by a special MAX/MIN button we already have.
So, to set necessary number, we should always delete current with "Backspace". 

Proposal:
Can we delete the current quantity in this window, when pressing any digit, and enter that digit in one single key?
Ex: on a screenshot we have 30 units, which on default were separated like 15/15. We need to separate 20. When we press "2" the "15"  disappears and "2" appears. Then we press "0" and all done!
Note, that it should happen when default separation dialog window appears, or after we touch any button in this window (Max, Min or arrows)

**Comment by LeHerosInconnu** [comment]
That's right, why bother deleting the previous number, just enter the desired number directly. :)
I suggest that it is also possible to use the numeric keypad to enter the desired number of creatures.

**Comment by ihhub** [comment]
Hi @Districh-ru , I think we might need to implement some sort of numpad for touch devices to cover this issue. This is a low priority so it is just food for thoughts🙂

**Comment by Districh-ru** [comment]
Sure)
